plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1547,6 +1550,43 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
     }
 
+    @Override
+    public void renameTable(ConnectorSession session, ConnectorTableHandle tableHandle, SchemaTableName newTableName)
+    {
+        DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
+        Table table = metastore.getTable(handle.getSchemaName(), handle.getTableName()).orElseThrow(() -> new TableNotFoundException(handle.getSchemaTableName()));
+
+        if (table.getTableType().equals(MANAGED_TABLE.name())) {
+            if (!renameManagedTablesEnabled) {
+                throw new TrinoException(NOT_SUPPORTED, "Renaming managed tables will move the data on the underlying storage system. Set 'delta.rename-managed-tables.enabled' to enable this feature.");
+            }
+
+            Database schema = metastore.getDatabase(newTableName.getSchemaName()).orElseThrow(() -> new SchemaNotFoundException(newTableName.getSchemaName()));
+            Optional<String> schemaLocation = getSchemaLocation(schema);
+            if (schemaLocation.isEmpty()) {
+                throw new TrinoException(NOT_SUPPORTED, "Target schema must have a location when renaming managed tables");
+            }
+            Path targetPath = new Path(schemaLocation.get(), newTableName.getTableName());
+            String location = targetPath.toString();
+            ensurePathExists(session, targetPath);
+
+            Table.Builder tableBuilder = Table.builder(table)
+                    .setDatabaseName(newTableName.getSchemaName())
+                    .setTableName(newTableName.getTableName())
+                    // Explicitly set the location property because hive metastore won't update it
+                    .setParameters(deltaTableProperties(session, location, false));
+
+            // Explicitly set the path in storage because hive metastore won't update it
+            // Leave the location as is to trigger rename of the table in hive metastore
+            tableBuilder.getStorageBuilder().setSerdeParameters(ImmutableMap.of(PATH_PROPERTY, location));
+
+            metastore.alterTable(session, handle.getSchemaTableName(), tableBuilder.build());
+        }
+        else {
+            metastore.renameTable(session, handle.getSchemaTableName(), newTableName);
```

Comment:
We seem to not have this test covered in `BaseDeltaLakeMinioConnectorTest`

Reply:
The external table test has been moved from the smoke test to `BaseDeltaLakeMinioConnectorTest`. Managed tables are already tested within the standard `BaseConnectorTest` tests. 

The particular scenario of creating a managed table in a scheme without a set location seems impossible. We can't modify/remove the location  or any properties of a scheme outside of the creation.